src/Sylius/Bundle/CoreBundle/Resources/config/services/promotion.xml

```diff
@@ -17,6 +17,12 @@
             <argument type="service" id="sylius.repository.promotion" />
         </service>
 
+        <service id="sylius.promotion_processor" class="Sylius\Component\Promotion\Processor\PromotionProcessor">
```

Comment:
Shouldn't we overwrite `sylius.active_promotions_provider` with `ActivePromotionsByChannelProvider` instead of overwriting `sylius.promotion_processor` changing its argument?